### ISSUE TITLE
Don't always force one-element-per-line when a sequence literal contains a comma after the last element

### ISSUE THREAD

**Issue by nathanielmanistaatgoogle** [google/yapf#561]
As of 0.21.0, yapf formats
```
_MY_CONSTANT = (3,4,5,6,7)
```
as
```
_MY_CONSTANT = (3, 4, 5, 6, 7)
```
and formats
```
_MY_CONSTANT = (3,4,5,6,7,)
```
as
```
_MY_CONSTANT = (
    3,
    4,
    5,
    6,
    7,
)
```
(the only difference between the two inputs is the comma after the last element). I'd much rather the latter form, which I prefer writing, were laid out much more compactly. If there's any reason why it shouldn't be compact, I'm having trouble seeing it.

(We use yapf for all the Python code in [gRPC Python](https://github.com/grpc/grpc/tree/master/src/python) and this is my biggest complaint, so congratulations on everything else about it being great!)

**Comment by jasonkuhrt** [comment]
I like having per-line items, arguments, etc. It allows me to make code that isn't 80 columns wide be consistent with code that exceeds 80 columns. Diffs can be nicer too. That said its not either-or, should be something that can be configured?

**Comment by nathanielmanistaatgoogle** [comment]
@jasonkuhrt: I don't think this needs configuration. You may be thinking of a separate switch to always have per-line items, but that would be outside the scope of this issue (unless you'd want your switch to behave differently based on whether or not there's a comma after the last element?).

**Comment by jasonkuhrt** [comment]
Maybe I'm misunderstanding. You would prefer that trialing comma does _not_ cause auto formatting one-element-per-line for _just_ sequences?

**Comment by gwelymernans** [comment]
This would need to be a separate knob I think. Let me see what I can do.

**Comment by gwelymernans** [comment]
I added a knob here. Could you test it out and see if it works for you?

To https://github.com/google/yapf.git
   ca82cbe..4922549  master -> master

**Comment by nathanielmanistaatgoogle** [comment]
[This commit](https://github.com/nathanielmanistaatgoogle/grpc/commit/b28c23e09b6f876b2662c716febfe08fdd106b42) is a first look at turning the knob - lots on which to follow up, now that I look directly at the size of it...

**Comment by gpshead** [comment]
The trailing comma hint is important.  **black** notably goes too far and _inserts a trailing comma_ when reformatting (danger! that can alter semantics of code if they haven't gotten it _exactly_ right).

A heuristic that would make sense here is that a trailing comma in a container on something that already fits on one line does not need to trigger the one per line expansion.
